backend/station_manager/app/src/main/java/com/StationManager/app/domain/MapManager.java

```diff
@@ -0,0 +1,222 @@
+package com.StationManager.app.domain;
+
+import com.StationManager.app.domain.client.Client;
+import com.StationManager.app.domain.train_station.Direction;
+import com.StationManager.app.domain.train_station.Segment;
+import com.StationManager.app.domain.train_station.TicketOffice;
+
+import java.awt.*;
+import java.util.*;
+import java.util.function.Function;
+
+public class MapManager {
+    private static Segment size;
+
+    public static void setSize(Segment newSize) {
+        size = newSize;
+    }
+
+    /**
+     * Assigns a client to ticket office based on their current position and ticket offices'
+     * locations.
+     *
+     * <p>This method determines the ticket office which is the most suitable for client to be
+     * assigned to. If there is one ticketOffice, client is assigned to this ticket office If there
+     * are more than one ticket offices, client is assigned to the one, whose next free queue
+     * position is closer to client
+     *
+     * @param client The client to be assigned to a ticket office.
+     * @param entrances The list of entrance segments.
+     * @param ticketOffices The list of ticket offices.
+     */
+    public static void assignClientToClosestTicketOffice(
```

Comment:
`entrances` are not used here

Reply:
`entrances` and `ticketOffices` in this case were used to check wether calculated new client position is free or not. Should we make it directly in this class or there are some other approaches?